index.js

```diff
@@ -82,35 +88,17 @@ function buildProcessTree (parentPid, tree, pidsToProcess, cb) {
             if (Object.keys(pidsToProcess).length == 0) {
                 cb();
             }
-            return
-        }
-
-        pids = [];
-        pid = '';
-        for (i = 0; i < allData.length; i++) {
-            if (allData[i] == '\n') {
-                pids.push(parseInt(pid, 10));
-                pid = '';
-                continue;
-            }
-            if (allData[i] != ' ') {
-                pid += allData[i];
-            }
```

Comment:
I believe it is less error prone to match sequence of digits with a simple regex (`/\d+/g`) than rolling your own `ps` output parser.